### PR TITLE
matomo integration

### PR DESCRIPTION
**Pull request by mohsenParvar**

### CODE REVIEW

File: components/UI/Buttons/LinkButton/index.js [hunk 1]
```diff
@@ -27,11 +30,20 @@ const LinkButton = ({
 }) => {
   const classes = useStyles();
 
+  const { trackEvent } = useAnalytics()
+
```
Comment (saito-sv) [inline components/UI/Buttons/LinkButton/index.js:34]: remove extra space

File: components/WalletModal/index.js [hunk 2]
```diff
@@ -10,6 +10,9 @@ import WalletCard from 'components/WalletModal/WalletCard'
 import { walletlink, injected } from 'utils/constants/connectors'
 import ANIMATIONS from 'utils/constants/animate-icons'
 
+import { AnalyticActions, AnalyticCategories, createEvent, useAnalytics } from "contexts/analytics";
+
+
```
Comment (saito-sv) [inline components/WalletModal/index.js]: remove extra space

File: contexts/analytics.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import { MatomoProvider, createInstance, useMatomo } from '@datapunt/matomo-tracker-react'
+import { useRouter } from "next/router";
+import { useEffect, useRef } from 'react'
+
+
```
Comment (saito-sv) [inline contexts/analytics.js]: remove extra space

File: pages/_app.js [hunk 4]
```diff
@@ -22,7 +23,13 @@ import { CompoundAndEarnProvider } from 'contexts/compound-and-earn-context'
 import { StakingContractProvider } from 'contexts/staking-context'
 import { ProviderProvider } from 'contexts/provider-context'
 
+
 function MyApp({ Component, pageProps }) {
+
+
+
+
+
```
Comment (saito-sv) [inline pages/_app.js]: remove extra spaces

File: parts/Compound/CompoundDialogs/index.js [hunk 5]
```diff
@@ -52,6 +53,9 @@ const useStyles = makeStyles(theme => ({
 
 const CompoundDialogs = ({ open, title, item, handleClose }) => {
   const classes = useStyles();
+
```
Comment (saito-sv) [inline parts/Compound/CompoundDialogs/index.js]: remove unnecessary extra spaces

File: parts/SubMenuTabs/index.js [hunk 6]
```diff
@@ -2,6 +2,8 @@
 import { memo } from 'react'
 import { makeStyles } from '@material-ui/core/styles'
 import { Button, Card } from '@material-ui/core'
+import { AnalyticActions, AnalyticCategories, createEvent, useAnalytics } from "contexts/analytics";
+
```
Comment (saito-sv) [inline parts/SubMenuTabs/index.js]: remove unnecessary space

File: contexts/analytics.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    name, // optional
+    value, // optional, numerical value
+    documentTitle, // optional
+    href, // optional
+  }
+}
+
+const urlBase = '//analytics.snowapi.net/'
+
+const instance = createInstance({
```
Comment (saito-sv) [inline contexts/analytics.js:54]: lets give this const a better name please

File: contexts/analytics.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    category,
+    action,
+    name, // optional
+    value, // optional, numerical value
+    documentTitle, // optional
+    href, // optional
+  }
+}
+
+const urlBase = '//analytics.snowapi.net/'
```
Comment (saito-sv) [inline contexts/analytics.js]: use .env file to hold this value

**Review by saito-sv — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
How is the state of this? How I can test it?

**Review by saito-sv — APPROVED** [review]
LGTM

**Comment by saito-sv** [comment]
> How is the state of this? How I can test it?

@Jonasslv  should be good to go.

**Pull request merged**